### PR TITLE
Implement a variety of essential standard library and language features

### PR DESCRIPTION
**Pull request by piotrwyrw**
This patch introduces the `lgt` and `rgt` functions for value comparison alongside their syntax bindings `>` and `<`. These functions perform either numerical, lexical order or string length comparison depending on which one is the best option. More details on that are in the new README. This also adds the string length ("_magnitude_") operator `| expr |` and its backing function `len`. As a final touch, the _string conversion operator_, a pure lexer feature, was introduced. This operator converts every token that follows to a string literal token.

### CODE REVIEW

File: src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt [hunk 1]
```diff
@@ -49,6 +59,22 @@ class Standard(val runtime: Runtime) {
...
+    fun parameter(id: Node): String = runtime.parameters[id.evaluate(runtime)] ?: ""
+
+    @DiscreteFunction(identifier = "set", params = ["key", "value"])
+    fun set(key: Node, value: Node): String {
+        variableStore[key.evaluate(runtime)] = value.evaluate(runtime)
+        return ""
+    }
+
+    @DiscreteFunction(identifier = "get", params = ["key"])
+    fun set(key: Node): String = variableStore[key.evaluate(runtime)] ?: ""
```
Comment (RastiPatak) [inline src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt:72]: isn't this meant to be ```get(key: Node)``` ?

File: src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt [hunk 2]
```diff
@@ -60,20 +86,55 @@ class Standard(val runtime: Runtime) {
         return _else.evaluate(runtime)
     }
 
-    @DiscreteFunction(identifier = "equal", params = ["a", "b"])
+    @DiscreteFunction(identifier = "equal", params = [CMP_LEFT, CMP_RIGHT])
     fun equal(a: Node, b: Node): String = if (a.evaluate(runtime) == b.evaluate(runtime)) "true" else "false"
```
Comment (RastiPatak) [inline src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt:90]: does this work if you change ```params```, but don't change the parameters in ```equal()```?

File: src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt (hunk 1 above)
Comment (piotrwyrw) [reply src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt]: White it _technically_ wouldn't matter, as the `LibraryManager`'s `invokeLibMethod` function only _really_ cares about the annotation when looking for functions (below an excerpt from said function) ...
```kotlin
val rawMethods =
            lib.declaredMethods.filter { it.getAnnotation(DiscreteFunction::class.java)?.identifier == name }
```
.. it definitely is very confusing to read and something that I have missed. Thanks for pointing it out! Resolved in 44317c1.

File: src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt (hunk 2 above)
Comment (piotrwyrw) [reply src/main/kotlin/org/medaware/anterogradia/libs/Standard.kt]: Yep! It does! Again, the reflections engine is pretty flexible when it comes to naming things. But again, I'll totally agree with your point of view that it is quite confusing. Thanks!
Resolved in 289b87b.

**Review by RastiPatak — APPROVED** [review]

**Pull request merged**